docs/concepts/workers.md

```diff
@@ -4,4 +4,52 @@ title: Workers
 sidebar_label: Workers
 ---
 
-A Worker is a service that hosts the [Workflow](/docs/concepts/workflows) and [Activity](/docs/concepts/activities) implementations.
+A Worker is a service that does the following:
+
+- Hosts executable [Workflow](/docs/concepts/workflows) and [Activity](/docs/concepts/activities) code.
+- Listens to [Task Queues](/docs/concepts/task-queues) via long polling.
+
+Workers must have access to any resources needed to execute the actions defined in Activities, such as the following:
```

Comment:
This is a little confusing to me, does this mean i have to provide all my credentials to the worker when I register one? I'm not sure how to read / interpret these rules.

Reply:
just for the activities that you want to execute?

Reply:
oh i see, yes you can pass parameters to activities when you register them like

worker.registerActivitiesImplementations(new GreetingActivitiesImpl("mySecretString));

I think in some cases users would use secrets or env vars instead of passing strings to activities..maybe it would be ok to put that as optional to not confuse users that they must do this ? just wondering

Reply:
i didn't say anything about *how* the credentials are passed - it can be as params, it can be as env vars :)